File: Easy/Branch Sums/Branch Sums.cpp
#include <vector>
using namespace std;

class BinaryTree
{
public:
    int value;
    BinaryTree *left;
    BinaryTree *right;

    BinaryTree(int value)
    {
        this->value = value;
        left = nullptr;
        right = nullptr;
    }
};

void helper(BinaryTree *node, int sum, vector<int> &output)
{
    if (node == nullptr)
        return;

    sum += node->value;
    if (node->left == nullptr && node->right == nullptr)
    {
        output.push_back(sum);
        return;
    }

    helper(node->left, sum, output);
    helper(node->right, sum, output);
}

vector<int> branchSums(BinaryTree *root)
{
    vector<int> output;
    helper(root, 0, output);
    return output;
}

File: Easy/Branch Sums/Branch Sums.py
class BinaryTree:
    def __init__(self, value):
        self.value = value
        self.left = None
        self.right = None


def helper(node: BinaryTree, branch, array):
    if node != None:
        branch += node.value
        if node.left == None and node.right == None:
            array.append(branch)
            return
        helper(node.left, branch, array)
        helper(node.right, branch, array)


def branchSums(root):
    array = []
    helper(root, 0, array)
    return array
